Extension/src/common.ts

```diff
@@ -165,19 +165,19 @@ export function getVcpkgRoot(): string {
 export function isHeaderFile(uri: vscode.Uri): boolean {
     const fileExt: string = path.extname(uri.fsPath);
     const fileExtLower: string = fileExt.toLowerCase();
-    return !fileExt || [".cuh", ".hpp", ".hh", ".hxx", ".h++", ".hp", ".h", ".ii", ".inl", ".idl", ""].some(ext => fileExtLower === ext);
+    return !fileExt || [".cuh", ".hpp", ".hh", ".hxx", ".h++", ".hp", ".h", ".inl", ".ipp", ".tcc", ".tlh", ".tli", ""].some(ext => fileExtLower === ext);
 }
 
 export function isCppFile(uri: vscode.Uri): boolean {
     const fileExt: string = path.extname(uri.fsPath);
     const fileExtLower: string = fileExt.toLowerCase();
-    return (fileExt === ".C") || [".cu", ".cpp", ".cc", ".cxx", ".c++", ".cp", ".ino", ".ipp", ".tcc"].some(ext => fileExtLower === ext);
+    return (fileExt === ".C") || [".cu", ".cpp", ".cc", ".cxx", ".c++", ".cp", ".ii", ".ino"].some(ext => fileExtLower === ext);
 }
 
 export function isCFile(uri: vscode.Uri): boolean {
     const fileExt: string = path.extname(uri.fsPath);
     const fileExtLower: string = fileExt.toLowerCase();
-    return (fileExt === ".C") || fileExtLower === ".c";
+    return fileExt === ".c" || fileExtLower === ".i";
```

Comment:
Is `.I` supposed to be a C file?

Reply:
Similarly, are all the other extension supposed to be case-insensitive?

Reply:
Should the extensions be case sensitive on Linux/Mac and case insensitive on Windows?

Reply:
".i" and ".ii" are compiler-generated preprocessed file.  (And Bob pointed out that MSVC broke that rule and is outputting ".i" for all preprocessed output, unlike GCC and Clang).  I don't think there is any precedent uppercase ".I" have a special meaning.

Reply:
VS Code itself considers ".i" to be a C source files.  We had it configured as a C source file on the native side already.

Reply:
But this code is treating ".I" as a C file, when on Linux it would not normally be due to case sensitivity, right?

Reply:
The file associations from VS Code itself are case insensitive.  We have this comment in the native code: 

`// VS Code file associations are case-insensitive on all OS's.`

(And some research seems to support that).

I think the right approach would be to consider all file associations as being case insensitive, with the uppercase `.C` being the only exception.

Reply:
ok